website_compassion/template/my_account_donations.xml

```diff
@@ -149,7 +150,34 @@
                         data-target="#modal_payment_options">Change</button>
             </div>
         </template>
-
+        <template id="my_sponsorship_donation">
+            <div class="mt-4">
+                <!-- Payments option -->
+                <h3>My sponsorships</h3>
+                <div class="row" style="backgroud: lightgray;">
+                    <div class="col-6">
+                        <label><b>Child</b></label>
+                    </div>
+                    <div class="col-3">
+                        <label><b>Amount</b></label>
+                    </div>
+                </div>
+                <t t-foreach="paid_sponsor" t-as="sponsor">
+                    <div class="row">
+                        <div class="col-6">
+                            <label><t t-esc="sponsor.child_id.preferred_name"/></label>
+                        </div>
+                        <div class="col-3">
+                            <label><t t-raw="int(sponsor.total_amount)"/> <t t-esc="sponsor.invoice_line_ids[0].currency_id.name"/></label>
+                        </div>
+                        <div class="col-3" t-if="int(sponsor.total_amount) &lt;= 42">
+                          <a t-attf-href="/my/donations/update?recurring_contract=#{sponsor.id}" class="btn btn-primary">Upgrade to 50 <t t-esc="sponsor.invoice_line_ids[0].currency_id.name"/></a>
```

Comment:
If selected sponsorship < 42 CHF the upgrade wouldn't be all the way to 50 (with current implementation). 
Maybe a bit confusing for the end user ?